### PR TITLE
Pbench clear results

### PR DESCRIPTION
**Pull request by zulcss**
Pythonize pbench-clear-results

### CODE REVIEW

**Review by dbutenhof — APPROVED** [review]
Did we really need to have #1842 again? I skimmed over the code I recognized; I hope it's really the same. ;-)

Although I don't really understand the "tmp" and "tools" prefix exclusion, this looks like an accurate transcription of the bash... which I hope is correct!

File: lib/pbench/cli/agent/commands/results/clear.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    def execute(self):
+        if not self.pbench_run.exists():
+            LOG.warn("Unable to cleanup the '%s' directory", self.pbench_run)
+            return 1
+
+        for path in self.pbench_run.iterdir():
+            # NOTE WELL: we remove each entry explicitly using the full ${pbench_run}
+            # path just in case some bug inadvertently places us in the wrong
+            # diretory.
```
Comment (dbutenhof) [inline lib/pbench/cli/agent/commands/results/clear.py:36]: OK, you copied the bash script comment, but...

The reference to ${pbench_run} is inaccurate and a bit out of place in the Python translation.

Also, on the last line, "directory" was misspelled and should be fixed.

**Review by dbutenhof — COMMENTED** [review]
If this is really ready for review, consider promoting it out of "Draft" status?

File: lib/pbench/cli/agent/commands/results/clear.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            if not str(path.name).startswith("tmp") and not str(path.name).startswith(
```
Comment (portante) [inline lib/pbench/cli/agent/commands/results/clear.py:37]: I think this can be rewritten as:

```
if not str(path.name).startswith(("tmp", "tools")):
```

**Review by portante — CHANGES_REQUESTED** [review]

File: lib/pbench/cli/agent/commands/results/clear.py (hunk 2 above)
Comment (zulcss) [reply lib/pbench/cli/agent/commands/results/clear.py]: fixed

File: lib/pbench/cli/agent/commands/results/clear.py (hunk 1 above)
Comment (zulcss) [reply lib/pbench/cli/agent/commands/results/clear.py]: fixed

**Review by dbutenhof — DISMISSED** [review]
Looks good.

File: lib/pbench/cli/agent/commands/results/clear.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                # pbench_run path just in case some bug inadvertently places
+                # us in the wrong directory.
+                if not str(path.name).startswith(("tmp", "tools")):
+                    if path.is_dir():
+                        shutil.rmtree(path)
+                    else:
+                        path.unlink()
+            return 0
+        except Exception:
+            LOG.warn("Unable to cleanup the '%s' directory", self.pbench_run)
```
Comment (portante) [inline lib/pbench/cli/agent/commands/results/clear.py:41]: Perhaps include the string representation of the exception?  Something like: `LOG.warn("Unable to clean up the '%s' directory, '%s'", self.pbench_run, exc)`?

**Review by portante — CHANGES_REQUESTED** [review]
Looks good, can you also try performing a `make rpm` in the `agent/rpm` directory?  I think we'll need to add this command to those installed via the `agent/Makefile`.

File: lib/pbench/cli/agent/commands/results/clear.py (hunk 3 above)
Comment (zulcss) [reply lib/pbench/cli/agent/commands/results/clear.py]: done

**Review by dbutenhof — DISMISSED** [review]

**Review by riya-17 — APPROVED** [review]

File: agent/Makefile [hunk 4]
```diff
@@ -255,7 +255,6 @@ install-python3-setup: install-util-scripts install-lib
 	mkdir -p ${DESTDIR}/python3
 	(cd ..; SKIP_GENERATE_AUTHORS=1 SKIP_WRITE_GIT_CHANGELOG=1 python3 setup.py install --prefix=${DESTDIR}/python3)
 	${COPY} ${DESTDIR}/python3/bin/pbench-config ${UTILDIR}/
```
[the remark this answers is not in the document]
Comment (portante) [reply agent/Makefile:257]: It seems like we need to restore `pbench-clear-tools` and add `pbench-clear-results`.

**Review by portante — CHANGES_REQUESTED** [review]
Close ...

File: agent/Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (zulcss) [reply agent/Makefile:256]: done

**Review by dbutenhof — DISMISSED** [review]

**Review by portante — CHANGES_REQUESTED** [review]
Can we add a unit test for the `--help` and for just invoking it once?

**Review by riya-17 — DISMISSED** [review]

**Review by portante — APPROVED** [review]

**Review by dbutenhof — APPROVED** [review]

**Pull request merged**